src/FileType_XDMF.f90

```diff
@@ -0,0 +1,1324 @@
+          CALL eXDMF%raiseError(modName//'::'//myName//&
+            ' - set data in h5 file is the wrong size or shape.')
+        ENDIF
+        ! Data type
+        dtype=h5%getDataType(CHAR(group))
+        IF(dtype == 'SNK') THEN
+          CALL h5%fread(CHAR(group),ivals4_1d)
+        ELSE
+          CALL h5%fread(CHAR(group),ivals8_1d)
+        ENDIF
+        ! Resize cell sets if needed
+        ! This is expected to happen infrequently
+        IF(ALLOCATED(mesh%cell_sets)) THEN
+          ! Copy current sets to temp, deallocate current sets
+          ncell_sets = SIZE(mesh%cell_sets)
+          ALLOCATE(cell_sets_temp(ncell_sets))
+          DO j=1, ncell_sets
+            ALLOCATE(cell_sets_temp(j)%cell_list(SIZE(mesh%cell_sets(j)%cell_list)))
+            cell_sets_temp(j)%cell_list = mesh%cell_sets(j)%cell_list
+            cell_sets_temp(j)%name = mesh%cell_sets(j)%name
+            DEALLOCATE(mesh%cell_sets(j)%cell_list)
+          ENDDO
+          DEALLOCATE(mesh%cell_sets)
+          ! Reallocate cell sets to be on bigger and copy all old sets over
+          ALLOCATE(mesh%cell_sets(ncell_sets+1))
+          DO j=1, ncell_sets
+            ALLOCATE(mesh%cell_sets(j)%cell_list(SIZE(cell_sets_temp(j)%cell_list)))
+            mesh%cell_sets(j)%cell_list = cell_sets_temp(j)%cell_list
+            mesh%cell_sets(j)%name = cell_sets_temp(j)%name
+            DEALLOCATE(cell_sets_temp(j)%cell_list)
+            CALL cell_sets_temp(j)%name%clear()
+          ENDDO
+          DEALLOCATE(cell_sets_temp)
+        ELSE
+          ncell_sets = 0
+          ALLOCATE(mesh%cell_sets(1))
+        ENDIF
+        ! Add the one new cell set
+        mesh%cell_sets(ncell_sets + 1)%name = elname
+        ALLOCATE(mesh%cell_sets(ncell_sets + 1)%cell_list(ncells))
+        IF(dtype == 'SNK') THEN
+          ! Account for 0 based to 1 based index switch
+          mesh%cell_sets(ncell_sets + 1)%cell_list = ivals4_1d + 1
+          DEALLOCATE(ivals4_1d)
+        ELSE
+          mesh%cell_sets(ncell_sets + 1)%cell_list = ivals8_1d + 1
+          DEALLOCATE(ivals8_1d)
+        ENDIF
+
+      CASE DEFAULT
+        CALL eXDMF%raiseWarning(modName//'::'//myName// &
+          ' - Unsupported data in XDMF file '//CHAR(elname))
+    ENDSELECT
+  ENDDO
+ENDSUBROUTINE setup_leaf_XDMFMesh_from_file
+!
+!-------------------------------------------------------------------------------
+!> @brief Imports the mesh data in the file to a mesh object.
+!> @param thisXDMFFile the XDMF file type object
+!> @param strpath the string holding the path to the XDMF file
+!> @param mesh the XDMF mesh object
+!>
+SUBROUTINE importFromDisk_XDMFFileType(thisXDMFFile, strpath, mesh)
+  CHARACTER(LEN=*),PARAMETER :: myName='importFromDisk_XDMFFileType'
+  CLASS(XDMFFileType),INTENT(INOUT) :: thisXDMFFile
+  TYPE(StringType),INTENT(INOUT) :: strpath
+  TYPE(XDMFMeshType),INTENT(OUT),TARGET  :: mesh
+  TYPE(XMLFileType) :: xml
+  TYPE(HDF5FileType) :: h5
```

Comment:
Looks like you are reading in multiple files to construct the XDMF data structure so that suggests again it shouldn't be derived from the BaseFileType.